shell/plugins/dashboard-store/__tests__/actions.spec.js

```diff
@@ -0,0 +1,125 @@
+import _actions from '@shell/plugins/dashboard-store/actions';
+
+const { findAll } = _actions;
+
+describe('dashboard-store: actions', () => {
+  let dispatchCalls = {};
+  let commitCalls = {};
+  // we're dumping our commits into a map so we can reference them
+  // actions and mutations are stored against their name in an array so we can capture if they're called multiple times
+  // arguments past the name of the action or mutation are stored in an array in case there are multiple
+  const commit = (mutation, ...args) => {
+    commitCalls[mutation] = [...(commitCalls[mutation] || []), args];
+  };
+  const dispatch = (action, ...args) => {
+    dispatchCalls[action] = [...(dispatchCalls[action] || []), args];
+    if (action === 'request') {
+      return { data: [] };
+    }
+  };
+  const state = { config: { namespace: 'unitTest' } };
+  const getters = {
+    normalizeType:    (type) => type,
+    typeRegistered:   (type) => type === 'foo',
+    haveAll:          (type) => type === 'baz',
+    haveAllNamespace: (type, namespaced) => type === 'far' && namespaced === 'faz',
+    all:              (type) => [type],
+    urlFor:           (type, _, opt) => type, // we're not testing the urlFor getter so we don't need to do anything with opt here
+  };
+  const rootGetters = {
+    'type-map/optionsFor': (type) => {
+      return {};
+    },
+    'auth/fromHeader': 'foo'
+  };
+
+  // we're not testing function output based off of state or getter inputs here since they are dependencies and should be tested independently
+  const ctx = {
+    state,
+    getters,
+    rootGetters,
+    commit,
+    dispatch
+  };
+
+  // setting afterEach here seems reasonable but isn't allowed by the linter
+  const resetCalls = () => {
```

Comment:
This is the outcome of not using the `beforeEach` as assignment part.

Reply:
this is actually the result of our jest unit test flagging the use of beforeEach and afterEach for setup and teardown of tests. I did go ahead and change this to hopefully more closely align with the suggested implementation in the Jest documentation but beforeEach isn't recommended in this case as it would require me to fully call "findAll" between which assertion for which we make quite a few and will probably need even more before we've fully covered it.

https://github.com/jest-community/eslint-plugin-jest/blob/v24.4.0/docs/rules/no-hooks.md

Reply:
As I tried to explain, that rule suggest to avoid both, the hook and external definitions. The second case is worse because it can be mutated and not be deterministic.
If you are not fine with the rule, you can always disable for the file or you can propose to remove it entirely.

Said that, any setup is more comfortable for your debugging is welcome 👍